data/transactions/logic/eval.go

```diff
@@ -1699,6 +1699,36 @@ func opDig(cx *evalContext) {
 	cx.stack = append(cx.stack, sv)
 }
 
+func opCover(cx *evalContext) {
+	depth := int(cx.program[cx.pc+1])
+	topIdx := len(cx.stack) - 1
+	idx := topIdx - depth
+	// Need to check stack size explicitly here because checkArgs() doesn't understand cover
+	// so we can't expect our stack to be prechecked.
+	if idx < 0 {
+		cx.err = fmt.Errorf("cover %d with stack size = %d", depth, len(cx.stack))
+		return
+	}
+	sv := cx.stack[topIdx]
+	copy(cx.stack[idx+1:], cx.stack[idx:])
+	cx.stack[idx] = sv
+}
+
+func opUncover(cx *evalContext) {
+	depth := int(cx.program[cx.pc+1])
+	idx := len(cx.stack) - 1 - depth
+	// Need to check stack size explicitly here because checkArgs() doesn't understand uncover
+	// so we can't expect our stack to be prechecked.
+	if idx < 0 {
+		cx.err = fmt.Errorf("uncover %d with stack size = %d", depth, len(cx.stack))
+		return
+	}
+	topIdx := len(cx.stack) - 1
```

Comment:
make it the same as prev function
```suggestion
	depth := int(cx.program[cx.pc+1])
	topIdx := len(cx.stack) - 1
	idx := topIdx - depth
	// Need to check stack size explicitly here because checkArgs() doesn't understand uncover
	// so we can't expect our stack to be prechecked.
	if idx < 0 {
		cx.err = fmt.Errorf("uncover %d with stack size = %d", depth, len(cx.stack))
		return
	}
```